src/verifier/verifier.controller.ts

```diff
@@ -9,14 +10,43 @@ import { VerifierService } from './verifier.service';
 @ApiTags('verifier')
 export class VerifierController {
 
+    private rate_limit_list: { ip: string, count: number }[] = [];
+    private list_created_at: Date = new Date();
+
     constructor(private readonly verifierService: VerifierService) {}
 
+    private checkAndLogIp(ipAddress: string): void {
+        // step 1: we are interested in the number of requests over time.  so
+        // check how long its been since we started collecting Ips and reset the list
+        // if the time span has been exceeded.
```

Comment:
Do we want the time span to be since collecting any IP or since collecting a specific IP?

Reply:
sure.   30 requests over a day is not a problem.  30 request for 30 seconds might be

Reply:
Right, that makes sense. So it should be since collecting a specific IP.